MoreLinq.Test/ScanTest.cs

```diff
@@ -43,6 +43,15 @@ public void ScanIsLazy()
             new BreakingSequence<object>().Scan(delegate { throw new NotImplementedException(); });
         }
 
+        [Test]
```

Comment:
The `Enumerable.Range(1, 3).Concat(new BreakingSequence<int>()).Scan(SampleData.Plus)` is repeated twice. Can we consolidate such duplication? For example:

```c#
        [Test]
        public void ScanDoesNotIterateExtra()
        {
            var xs =
                Enumerable.Range(1, 3)
                          .Concat(new BreakingSequence<int>())
                          .Scan(SampleData.Plus);
            Assert.Throws<InvalidOperationException>(() => xs.Consume());
            xs.Take(3).AssertSequenceEqual(1, 3, 6);
        }
```

Ditto for `SeededScanDoesNotIterateExtra`.

Reply:
fixed